.github/workflows/ci.yml

```diff
@@ -60,3 +60,50 @@ jobs:
           if [ ! -d docs/doxygen/ ]; then exit 0; fi
           doxygen docs/doxygen/config.doxyfile 2>&1 | tee doxyoutput.txt
           if [[ "$(wc -c < doxyoutput.txt | bc)" = "0" ]]; then exit 0; else exit 1; fi
+  spell-check:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout Parent Repo
+          uses: actions/checkout@v2
+          with:
+            ref: master
+            repository: aws/aws-iot-device-sdk-embedded-C
+        - run: rm -r libraries/standard/coreJSON
+        - name: Clone This Repo
+          uses: actions/checkout@v2
+          with:
+            path: libraries/standard/coreJSON
+        - name: Install spell
+          run: |
+            sudo apt-get install spell
+            sudo apt-get install util-linux
+        - name: Check spelling
+          run: |
+            PATH=$PATH:$PWD/tools/spell
+            for lexfile in `find libraries/standard/coreJSON -name lexicon.txt`
+            do dir=${lexfile%/lexicon.txt}
+              echo $dir
+              find-unknown-comment-words --directory $dir
+              if [ $? -ne "0" ]
+              then
+                exit 1
+              fi
+            done
+  formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Uncrustify
+        run: sudo apt-get install uncrustify
+      - name: Run Uncrustify
+        run: find . -iname "*.[hc]" -exec uncrustify --check -c tools/uncrustify.cfg {} +
+      - name: Check For Trailing Whitespace
```

Comment:
Why do we need this separately?  Does uncrustify above not check it?

Reply:
Uncrustify will do it just fine for source files, but we still have cmake files to check. Interestingly, this caught a trailing whitespace in one of them.